src/dashboard/src/components/forms/text/Text.tsx

```diff
@@ -32,8 +32,16 @@ export const Text: React.FC<TextProps> = ({
   iconType,
   className = '',
   disabled,
+  onChange,
+  value: initValue = '',
   ...rest
 }) => {
+  const [value, setValue] = React.useState(initValue);
```

Comment:
Need to maintain state within to improve performance.  React is horrible when using controlled inputs.